.github/workflows/typespec-validation-all.yaml

```diff
@@ -0,0 +1,69 @@
+name: TypeSpec Validation All
+
+on:
+  push:
+    branches:
+      - main
+      - RPSaaSMaster
+      - typespec-next
+
+  pull_request:
+    branches:
+      - main
+      - RPSaaSMaster
+      - typespec-next
+    paths:
+      - .gitattributes
+      - .prettierrc.json
+      - package-lock.json
+      - package.json
+      - tsconfig.json
+      - eng/**
+      - specification/suppressions.yaml
+      - specification/common-types/**
+
+      # Workflow and workflow dependencies
+      - .github/workflows/typespec-validation-all.yaml
+      - .github/actions/setup-node-npm-ci/**
+
+  schedule:
+    # Run 4x/day
+    - cron: '0 0,6,12,18 * * * '
+
+jobs:
+  typespec-validation-all:
+    name: TypeSpec Validation All
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        # shards must start at 0 and increment by 1
+        shard: [0, 1, 2]
+        # total-shards must be an accurate count of the number of shards
+        total-shards: [3]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Enable git long paths
+        if: runner.os == 'Windows'
+        run: git config --global core.longpaths true
+
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+
+      - name: Setup Node 20 and run `npm ci`
+        uses: ./.github/actions/setup-node-npm-ci
```

Comment:
@mikeharder @danieljurek I thought github force these to be under the workflows directory. Is this a way we can make other reusable pieces for actions?

Reply:
The workflow itself must be under `.github/workflows`, but I believe custom actions can live anywhere in your repo, in another repo, etc.

We're still experimenting, but we're considering trying to provide most reusable code through custom actions, rather than using `github-script` everywhere and sharing JS code.